xbuild/src/gradle/mod.rs

```diff
@@ -67,6 +67,36 @@ pub fn build(env: &BuildEnv, libraries: Vec<(Target, PathBuf)>, out: &Path) -> R
         dependencies.push_str(&format!("implementation '{}'\n", dep));
     }
 
+    let mut dexes = String::new();
+    for dex in &env.config().android().dexes {
+        let mut path = env.cargo().package_root().join(dex);
+
+        // Pop the filename and use the directory.
+        //
+        // This is needed as we must provide a directory to `DexMergingTask::dexDirs`
+        path.pop();
+
+        let path = path.display().to_string().replace(r"\", r"/");
+
+        let external_lib = format!(r#"task.dexDirs.from("{path}")"#);
+        dexes.push_str(&external_lib);
+        dexes.push_str("\n");
+    }
+
+    let dexes = if !dexes.is_empty() {
+        format!(
+            r#"
+            afterEvaluate {{
+                tasks.named("mergeDexRelease").configure {{ task ->
```

Comment:
Sounds like this breaks if we build debug. As discussed and shown in-person yesterday, why not use `tasks.withType(TheTypeThatMergeDex{}Uses)`?

Reply:
> Sounds like this breaks if we build debug.

Potentially but not sure yet. In evolve we're not building AAB in debug ever so I decided to stick with this solution for now. That being said, I do agree that it should also work on debug at some point!

> As discussed and shown in-person yesterday, why not use tasks.withType(TheTypeThatMergeDex{}Uses)?

That is incorrect also as you could potentially add the directory to multiple tasks and you don't even really know which tasks. I prefer to be explicit and specify one task.

Reply:
By that logic there could be multiple `mergeDexRelease/Debug` tasks in the chain, making this equally incorrect :upside_down_face: